src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -163,109 +163,99 @@ public SimpleQueryStringBuilder flags(SimpleQueryStringFlag... flags) {
         return this;
     }
 
+    /**
+     * Specify whether expanded terms from a wildcard should be
+     * automatically lowercased.
+     */
     public SimpleQueryStringBuilder lowercaseExpandedTerms(boolean lowercaseExpandedTerms) {
         this.lowercaseExpandedTerms = lowercaseExpandedTerms;
         return this;
     }
 
+    /**
+     * Specify the locale to use for lowercasing expanded terms
+     */
     public SimpleQueryStringBuilder locale(Locale locale) {
         this.locale = locale;
         return this;
     }
 
+    /**
+     * Specify whether the parser should treat numeric values leniently
+     */
     public SimpleQueryStringBuilder lenient(boolean lenient) {
         this.lenient = lenient;
         return this;
     }
 
+    /**
+     * Specify whether wildcards ('*' and '?') should be analyzed
+     */
     public SimpleQueryStringBuilder analyzeWildcard(boolean analyzeWildcard) {
         this.analyzeWildcard = analyzeWildcard;
         return this;
     }
 
+    /**
+     * Specify the number of clauses that must match
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
     @Override
-    public void doXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject(SimpleQueryStringBuilder.NAME);
+    public String[] names() {
+        return new String[]{NAME, Strings.toCamelCase(NAME)};
+    }
 
-        builder.field("query", queryText);
+    @Override
+    public Query parse(QueryParseContext parseContext)
+            throws IOException, QueryParsingException {
+        SimpleQueryStringBuilder sqsb = new SimpleQueryStringBuilder();
+        sqsb.fromXContent(parseContext);
+        return sqsb.toQuery(parseContext);
+    }
 
-        if (fields.size() > 0) {
-            builder.startArray("fields");
-            for (Map.Entry<String, Float> entry : fields.entrySet()) {
-                String field = entry.getKey();
-                Float boost = entry.getValue();
-                if (boost != null) {
-                    builder.value(field + "^" + boost);
-                } else {
-                    builder.value(field);
-                }
+    public Query toQuery(QueryParseContext parseContext) {
+        Analyzer analyzer;
+        // Use standard analyzer by default if none specified
+        if (analyzerName == null) {
+            analyzer = parseContext.mapperService().searchAnalyzer();
+        } else {
+            analyzer = parseContext.analysisService().analyzer(analyzerName);
+            if (analyzer == null) {
+                // TODO should we try to do this in fromXContent also, to make
+                // sure we can access the analyzer sooner?
```

Comment:
yea I think this validation should be done on the coordinating node ideally, rather sooner than later. That said we might need to do the same on the data node too (toQuery). Shall we have a validate method or something that can be called from both?